### PR TITLE
Split `d` into explicit `w` and `v` utility functions

### LINKED ISSUE #162: Split 'd' into specific 'w' and 'v' implementations
**Enhancement**

The inconsistency between the widget and Hyperscript params of `d` meant that it was difficult to sufficiently overload the API for ease of use. In addition splitting `d` into specific `w` and `v` implementations will enable the flexibility for extension.

Package Version: latest

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

As per the issue, the `d` implementation was becoming too overloaded and as a result not user friendly for consumers of the API. The intellisense and typings were getting very difficult to maintain across what we essentially two completely different parameter signatures.

Splitting `d` into discrete functions also enables future enhancements to be applied with low risk.

Resolves #162

### CODE REVIEW

**Comment by rishson** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -135,12 +136,14 @@ const myBasicWidget = createWidgetBase({
 
 #### `d`
 
-`d` is the canonical mechanism for `dojo-widgets` to express a widget hierarchical structure using either Dojo widget factories or Hyperscript.
+`d` is the module used to express hierarchical widget structures within Dojo 2. This structure constructed of `DNode`s (`DNode` is the intersection type of `HNode` and `WNode`).
+
+When creating custom widgets, you use the `v` and `w` functions from the `d` module.
```
Comment (Tomdye) [inline README.md]: drop `you`.

File: README.md [hunk 2]
```diff
@@ -149,36 +152,40 @@ The argument and return types are available from `dojo-widgets/interfaces` as fo
...
 Creates an element with the `tagName` with the `VNodeProperties` options and optional children specified by the array of `DNode`, `VNode` or `null`.
 
 ```ts
-d(tagName: string, options: VNodeProperties, children?: (DNode | VNode | null)[]): HNode[];
+v(tag: string, options: VNodeProperties, children?: (DNode | null)[]): HNode[];
 ```
-##### Dojo Widget
+##### `w`
+
+`w` is an abstraction layer for dojo-widgets that enables dojo 2's lazy instantiation, instance management and caching.
```
Comment (Tomdye) [inline README.md]: drop the apostrophe, no ownership.

**Comment by agubler** [comment]
Will pick the README up again after merging.

**Pull request merged**